rpcs3/Emu/Cell/Modules/cellGame.cpp

```diff
@@ -216,14 +234,107 @@ error_code cellHddGameCheck(ppu_thread& ppu, u32 version, vm::cptr<char> dirName
 
 	funcStat(ppu, result, get, set);
 
-	if (result->result != u32{CELL_HDDGAME_CBRESULT_OK} && result->result != u32{CELL_HDDGAME_CBRESULT_OK_CANCEL})
+	std::string error_msg;
+
+	switch (result->result)
+	{
+	case CELL_HDDGAME_CBRESULT_OK:
 	{
-		return CELL_HDDGAME_ERROR_CBRESULT;
+		// Game confirmed that it wants to create directory
+		const auto setParam = set->setParam;
+
+		if (new_data)
+		{
+			if (!setParam)
+			{
+				return CELL_GAMEDATA_ERROR_PARAM;
+			}
+
+			if (!fs::create_path(vfs::get(usrdir)))
```

Comment:
This function is called cellHddGameCheck, not cellHddGameCheckCreate. At best it only updates PARAM.SFO parameters with struct error checking for it.

Reply:
I do not think game are allowed to create HG game data, only GD.